File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
       ListNode sumList = new ListNode(0);
       ListNode headNode = sumList;
        int carry = 0;

       while (l1 != null || l2 != null || carry != 0) {
        int l1Val = l1 != null ? l1.val : 0;
        int l2Val = l2 != null ? l2.val : 0;
        int sum = l1Val + l2Val + carry;
        int modulo = sum % 10;
        carry = sum /  10;
        headNode.next = new ListNode(modulo);

        if (l1 != null) {
            l1 = l1.next;
        }
        if (l2 != null) {
            l2 = l2.next;
        }

        headNode = headNode.next;
       }

       return sumList.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        res = node = ListNode()
        carry = 0

        while l1 or l2:
            currSum = carry

            if l1:
                currSum += l1.val
                l1 = l1.next
            
            if l2:
                currSum += l2.val
                l2 = l2.next
            
            carry = currSum // 10
            node.next = ListNode(currSum % 10)
            node = node.next
        
        if carry:
            node.next = ListNode(carry)
        
        return res.next